internal/graph/check_fast_path.go

```diff
@@ -529,6 +544,7 @@ ConsumerLoop:
 }
 
 func constructLeftChannels(ctx context.Context,
+	logger logger.Logger,
```

Comment:
Perhaps bind the function to the struct so we don't have to pass this explicitly?